lisa/features/nvme.py

```diff
@@ -0,0 +1,79 @@
+# Copyright (c) Microsoft Corporation.
+# Licensed under the MIT license.
+
+import re
+from typing import Any, List
+
+from lisa.feature import Feature
+from lisa.tools import Lspci, Nvmecli
+from lisa.tools.lspci import PciDevice
+from lisa.util.logger import get_logger
+
+FEATURE_NAME_NVME = "NVME"
+
+
+class Nvme(Feature):
+    # crw------- 1 root root 251, 0 Jun 21 03:08 /dev/nvme0
+    _device_pattern = re.compile(r".*(?P<device_name>/dev/nvme[0-9]$)", re.MULTILINE)
+    # brw-rw---- 1 root disk 259, 0 Jun 21 03:08 /dev/nvme0n1
+    _namespace_pattern = re.compile(
+        r".*(?P<namespace>/dev/nvme[0-9]n[0-9]$)", re.MULTILINE
+    )
+    # '/dev/nvme0n1          351f1f720e5a00000001 Microsoft NVMe Direct Disk               1           0.00   B /   1.92  TB    512   B +  0 B   NVMDV001' # noqa: E501
+    _namespace_cli_pattern = re.compile(
+        r"(?P<namespace>/dev/nvme[0-9]n[0-9])", re.MULTILINE
+    )
+    _pci_device_name = "Non-Volatile memory controller"
+
+    def __init__(self, node: Any, platform: Any) -> None:
+        super().__init__(node, platform)
+        self._log = get_logger("feature", self.name(), self._node.log)
+
+    @classmethod
+    def name(cls) -> str:
+        return FEATURE_NAME_NVME
+
+    @classmethod
+    def enabled(cls) -> bool:
+        return True
+
+    @classmethod
+    def can_disable(cls) -> bool:
+        return True
+
+    def get_devices(self) -> List[str]:
+        devices = self._node.execute("ls -l /dev/nvme*", shell=True, sudo=True)
```

Comment:
Put the execute in a shared method, it can parse both device and namespaces together.